shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/compose/ContainerCompose.java

```diff
@@ -17,185 +17,105 @@
-        }
-        return null;
-    }
+    private final ImmutableList<ShardingSphereContainer> containers;
     
-    @SneakyThrows
-    private ShardingSphereContainer createContainer(final OnContainer metadata) {
-        switch (metadata.type()) {
-            case ADAPTER:
-                return createAdapterContainer();
-            case STORAGE:
-                return createStorageContainer();
-            case COORDINATOR:
-                throw new UnsupportedOperationException("");
-            default:
-                return null;
-        }
-    }
+    @Getter
+    private final ShardingSphereStorageContainer storageContainer;
     
-    private ShardingSphereAdapterContainer createAdapterContainer() {
-        switch (description.getAdapter()) {
-            case "proxy":
-                return new ShardingSphereProxyContainer();
-            case "jdbc":
-                return new ShardingSphereJDBCContainer();
-            default:
-                throw new RuntimeException("Adapter[" + description.getAdapter() + "] is unknown.");
-        }
-    }
+    @Getter
+    private final ShardingSphereAdapterContainer adapterContainer;
     
-    private ShardingSphereStorageContainer createStorageContainer() {
-        switch (description.getStorageType()) {
-            case MySQL:
-                return new MySQLContainer();
-            case H2:
-                return new H2Container();
-            default:
-                throw new RuntimeException("Unknown storage type " + description.getStorageType());
-        }
-    }
+    private volatile boolean started;
     
-    private void inject(final ShardingSphereContainer container) {
-        Collection<Field> fields = new LinkedList<>();
-        for (Class<?> klass = container.getClass(); Objects.nonNull(klass); klass = klass.getSuperclass()) {
-            fields.addAll(Arrays.asList(klass.getDeclaredFields()));
-        }
-        fields.stream().filter(each -> each.isAnnotationPresent(ShardingSphereITInject.class)).forEach(each -> setFieldValue(each, container));
+    public ContainerCompose(final String clusterName, final ParameterizedArray parameterizedArray) {
+        this.clusterName = clusterName;
+        this.parameterizedArray = parameterizedArray;
+        TestCaseDescription description = TestCaseDescription.builder()
+                .adapter(parameterizedArray.getAdapter())
+                .scenario(parameterizedArray.getScenario())
+                .database(parameterizedArray.getDatabaseType().getName())
+                .executionMode(ExecutionMode.SINGLE)
+                .sqlCommandType(SQLCommandType.DAL)
+                .build();
+        this.storageContainer = createStorageContainer(description);
+        this.adapterContainer = createAdapterContainer(description);
+        adapterContainer.dependsOn(storageContainer);
+        this.containers = ImmutableList.of(storageContainer, adapterContainer);
```

Comment:
Is it necessary to use ImmutableList here?
How about use raw JAVA with Arrays.asList?